pkg/ccl/crosscluster/logical/create_logical_replication_stmt.go

```diff
@@ -154,6 +157,10 @@ func createLogicalReplicationStreamPlanHook(
 					}
 					break
 				}
+				// The rowid column is always NOT NULL and is never explicitly set by the row processor, so ignore it
+				if requireNotNull && !col.Nullable && col.Name != "rowid" {
```

Comment:
the `col.Name != "rowid"` isn't the right check, as it assumes the pk is one column named "rowid". I _think_ the right check is 
```
requireNotNull && !col.Nullable && colNotInPrimaryKey(col.ID)
```

I just made up that last function, but you have the info required to create it. I would post in -eng about this, as I'm not totally confident this is the right check.